controllers/imageupdateautomation_controller.go

```diff
@@ -255,11 +256,19 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 		return failWithError(err)
 	}
 
-	lgc, err := libgit2pkg.NewClient(tmp, authOpts)
+	var gitClient git.RepositoryClient
+	switch origin.Spec.GitImplementation {
+	case libgit2.ClientName:
+		gitClient, err = libgit2.NewClient(tmp, authOpts)
+	case gogit.ClientName:
```

Comment:
The default empty behaviour should still be `go-git`:
```suggestion
	case gogit.ClientName, "":
```

Reply:
do you mean `libgit2`?

Reply:
Now we are getting the value from the `spec.gitImplementation`, I think we should ensure that it works in alignment with `source-controller`.

If in any scenario we need to force `libgit2` we can do that through other means.

Reply:
okay, since an intended side effect of this change is to reduce the number of crashes due to libgit2, i concur 👍

Reply:
Wouldn't it in general be better to match against the `v1beta2.GoGitImplementation` and `v1beta2.Libgit2Implementation` variables to ensure we match against the declarative API values opposed to the client package?